### PR TITLE
[libcxxwrap_julia] Add Julia_jll to libcxxwrap_julia

### PR DESCRIPTION
**Pull request by ViralBShah**
Not sure if this is sufficient or something more needs to be done.

Should it be a BuildDependency?

### CODE REVIEW

File: L/libcxxwrap_julia/build_tarballs.jl [hunk 1]
```diff
@@ -55,7 +55,7 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-
+	Dependency("Julia_jll")
```
Comment (giordano) [inline L/libcxxwrap_julia/build_tarballs.jl:58]: This should be a [`BuildDependency`](https://juliapackaging.github.io/BinaryBuilder.jl/dev/reference/#BinaryBuilder.BuildDependency):
```suggestion
    BuildDependency("Julia_jll")
```
Need also to remove all Julia sources above

File: L/libcxxwrap_julia/build_tarballs.jl [hunk 2]
```diff
@@ -12,10 +12,6 @@ unpack_target = is_yggdrasil ? "" : "libcxxwrap-julia"
 # Collection of sources required to complete build
 sources = [
     GitSource(git_repo, "7959f09947f95e67ee094a2ee3fb8b7f9be696f8", unpack_target=unpack_target),
-    ArchiveSource("https://julialang-s3.julialang.org/bin/linux/armv7l/1.3/julia-1.3.1-linux-armv7l.tar.gz", "965c8fab2214f8ce1b3d449d088561a6de61be42543b48c3bbadaed5b02bf824"; unpack_target="julia-arm-linux-gnueabihf"),
-    ArchiveSource("https://julialang-s3.julialang.org/bin/linux/x64/1.3/julia-1.3.1-linux-x86_64.tar.gz", "faa707c8343780a6fe5eaf13490355e8190acf8e2c189b9e7ecbddb0fa2643ad"; unpack_target="julia-x86_64-linux-gnu"),
-    ArchiveSource("https://github.com/Gnimuc/JuliaBuilder/releases/download/v1.3.0/julia-1.3.0-x86_64-apple-darwin14.tar.gz", "f2e5359f03314656c06e2a0a28a497f62e78f027dbe7f5155a5710b4914439b1"; unpack_target="julia-x86_64-apple-darwin14"),
-    ArchiveSource("https://github.com/Gnimuc/JuliaBuilder/releases/download/v1.3.0/julia-1.3.0-x86_64-w64-mingw32.tar.gz", "c7b2db68156150d0e882e98e39269301d7bf56660f4fc2e38ed2734a7a8d1551"; unpack_target="julia-x86_64-w64-mingw32"),
```
Comment (fingolfin) [inline L/libcxxwrap_julia/build_tarballs.jl:18]: There is some code below which hardcodes the version as 1.3.1, and which special cases ming32/darwin -- that needs to be adjusted.

Comment (fingolfin) [reply L/libcxxwrap_julia/build_tarballs.jl:18]: One could also enable the FreeBSD platform, I guess? Maybe hardcoding a list of platforms isn't necessary anymore at all?

**Comment by ViralBShah** [comment]
Julia_jll does not do powerpc or musl. So definitely don't have all the platforms.

**Comment by barche** [comment]
> Can someone take this PR over from me? I can leave the branch as is or close it...

I don't mind taking it over, for me it's easiest to just create a new PR since I'm not a member. BTW, how does this affect JLL versioning? Is it OK to push an update without changing the version?

**Comment by giordano** [comment]
> I don't mind taking it over, for me it's easiest to just create a new PR since I'm not a member

Are you sure? :raised_eyebrow: 

> BTW, how does this affect JLL versioning? Is it OK to push an update without changing the version?

Yes, the build version will be increased.  However, usually we use the same version number as upstream, but sincere here there isn't really an upstream...

**Comment by barche** [comment]
> Are you sure? raised_eyebrow

Ah, no, I was wrong 😂

I'll recycle this PR then and also bump the version, which is in sync with upstream, but 0.7.1 exists now.

**Comment by barche** [comment]
Should be good to go, note that I copied the platforms from Julia, and because these explicitly state the string ABI I also removed the `expand_cxxstring_abis`. Not sure if that's OK.

**Comment by giordano** [comment]
> and because these explicitly state the string ABI I also removed the expand_cxxstring_abis. Not sure if that's OK

It's necessary, more than OK.  If you try to build for C++03 string ABI you wouldn't get the binary for Julia.

**Pull request merged**